eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -1792,6 +1792,27 @@ default T detectIfNone(Predicate<? super T> predicate, Function0<? extends T> fu
         return Lists.immutable.withAllSorted(this);
     }
 
+    /**
+     * Converts the collection to an ImmutableList implementation and sorts it using the specified comparator.
+     *
+     * @since 11.0
+     */
+    default ImmutableList<T> toImmutableSortedList(Comparator<? super T> comparator)
+    {
+        return Lists.immutable.withAllSorted(comparator, this);
+    }
+
+    /**
+     * Converts the collection to a MutableList implementation and sorts it based on the natural order of the
```

Comment:
Minor comment - This should be ImmutableList in javadoc.

Reply:
Good catch, thank you! Will fix now.